torch/lib/THC/THCTensorScatterGather.cu

```diff
@@ -92,7 +92,7 @@ __global__ void THCudaTensor_gatherKernel(
                                                           tensor, &tensorOffset,
                                                           src, &srcOffset);
 
-    IndexType indexValue = (IndexType)index.data[indexOffset] - TH_INDEX_BASE;
+    long indexValue = index.data[indexOffset] - TH_INDEX_BASE;
```

Comment:
If the Tensor is "32-bit addressable", then its an unsigned int. Otherwise its an unsigned long.